src/mesh/mesh_refinement.cpp

```diff
@@ -787,6 +804,13 @@ void MeshRefinement::ProlongateSharedFieldX1(
         Real dx2p = x2p - x2c;
         const Real& fx2m = pco->x2s1(fj);
         const Real& fx2p = pco->x2s1(fj+1);
+        if (std::strcmp(COORDINATE_SYSTEM, "spherical_polar") == 0) {
```

Comment:
While modern compilers should be able to optimize it automatically, can you move this if statement out of the loop for better optimization?